### PR TITLE
Login Panel with callback

### PR DESCRIPTION
**Pull request by heyimcesar**
In this version I've created a Main Gui in order to handle our functions and the LoginPanel is generated with a simple interface and a callback to handle the login.

### CODE REVIEW

File: src/codeu/chat/client/maingui/LoginPanel.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    public LoginPanel(ClientContext clientContext, LoginInputCallback loginInputCallback) {
+        super();
+        this.clientContext = clientContext;
+        this.loginInputCallback = loginInputCallback;
+        initialize();
+    }
+    private void initialize() {
+
+        // This panel contains the login form for get access to the chat.
```
Comment (YoaliSB) [inline src/codeu/chat/client/maingui/LoginPanel.java:24]: Nitpick, consider revising grammar i.e, "This panel contains the login form to get access to the chat" or "This panel contains the login form to access the chat"

**Review by YoaliSB — APPROVED** [review]

File: src/codeu/chat/client/maingui/LoginPanel.java (hunk 1 above)
Comment (heyimcesar) [reply src/codeu/chat/client/maingui/LoginPanel.java:24]: True! I will merge the branch and in the next one I will correct it.

Comment (raguirre) [reply src/codeu/chat/client/maingui/LoginPanel.java:24]: Just a heads up, any kind of errors in the code should be corrected before merging. A grammar error in a comment is not a big deal but you don't ever want to merge "faulty" code to the upper branches.

Comment (wilph) [reply src/codeu/chat/client/maingui/LoginPanel.java:24]: BTW, yes, a more typical workflow here would be to resolve Yoali's comment first before completing the merge -- create a new commit that addresses review comments in login-main-view and push that back to github, that should automatically update the pull request.  (I think those are the steps based on http://stackoverflow.com/a/7947337, I've never used git+github).

Resolving a comment in a follow up merge sometimes makes sense if, e.g., the solution is big and out of scope of your change ("refactor this method" or something).

**Pull request merged**